internal/controller/controllers/controller_event_wrapper.go

```diff
@@ -58,13 +58,35 @@ func (c *controllerEventsWrapper) SendClusterEventAtTime(ctx context.Context, ev
 func (c *controllerEventsWrapper) SendHostEvent(ctx context.Context, event events.HostEvent) {
 	c.events.SendHostEvent(ctx, event)
 
-	c.NotifyKubeApiHostEvent(event.GetClusterId(), event.GetHostId())
+	if event.GetClusterId() == nil {
+		c.NotifyKubeApiHostEvent(event.GetInfraEnvId(), event.GetHostId())
+	} else {
+		c.NotifyKubeApiHostEvent(*event.GetClusterId(), event.GetHostId())
+	}
 }
 
 func (c *controllerEventsWrapper) SendHostEventAtTime(ctx context.Context, event events.HostEvent, eventTime time.Time) {
 	c.events.SendHostEventAtTime(ctx, event, eventTime)
 
-	c.NotifyKubeApiHostEvent(event.GetClusterId(), event.GetHostId())
+	if event.GetClusterId() == nil {
+		c.NotifyKubeApiHostEvent(event.GetInfraEnvId(), event.GetHostId())
+	} else {
+		c.NotifyKubeApiHostEvent(*event.GetClusterId(), event.GetHostId())
+	}
+}
+
+func (c *controllerEventsWrapper) SendInfraEnvEvent(ctx context.Context, event events.InfraEnvEvent) {
+	c.events.SendInfraEnvEvent(ctx, event)
+
+	// TODO: NotifyKubeApiInfraEnv
+	c.NotifyKubeApiClusterEvent(event.GetInfraEnvId())
```

Comment:
Need to notify only infra env 
suggesting to add some helper function here as well 
Same comment for `SendInfraEnvEventAtTime` 
```
func (c *controllerEventsWrapper) NotifyKubeApiInfraEnvEvent(infraEnvId strfmt.UUID) {
	ie, err := common.GetInfraEnvFromDB(c.db, infraEnvId)
	if err != nil {
		return
	}

	c.log.Debugf("Pushing InfraEnv event %s %s", ie.Name, ie.KubeKeyNamespace)
	c.crdEventsHandler.NotifyClusterDeploymentUpdates(ie.Name, ie.KubeKeyNamespace)
}
```
and use it here.

Reply:
done